services/frontend-service/src/ui/Pages/CommitInfo/CommitInfoPage.test.tsx

```diff
@@ -119,12 +139,55 @@ Commit message body line 2`,
                     nextCommitHash: '',
                 },
             },
+            expectedNumLoginPage: 0,
+        },
+        {
+            name: 'A login page renders when Dex is enabled',
+            fakeLoadEverything: true,
+            enableDex: true,
+            enableDexValidToken: false,
+            commitHash: 'potato',
+            expectedSpinnerCount: 0,
+            expectedMainContentCount: 0,
+            expectedText: 'Log in to Dex',
+            commitInfoStoreData: {
+                commitInfoReady: CommitInfoState.LOADING,
+                response: undefined,
+            },
+            expectedNumLoginPage: 1,
+        },
+        {
+            name: 'Some main content exists when Dex is enabled and the token is valid',
+            fakeLoadEverything: true,
+            enableDex: true,
+            enableDexValidToken: true,
+            commitHash: 'potato',
+            expectedSpinnerCount: 0,
+            expectedMainContentCount: 1,
+            expectedText: 'Add google to windows', // this "Commit + commit_message_first_line" string comes from the CommitInfo component logic (so we know that it actually rendered without having some mocking magic)
+            commitInfoStoreData: {
+                commitInfoReady: CommitInfoState.READY,
+                response: {
+                    commitHash: 'potato',
+                    touchedApps: ['google', 'windows'],
```

Comment:
nitpick: I think using "google" here as an example is a bit misleading, because this test has nothing to do with google.
```suggestion
                    touchedApps: ['firstApp', 'secondApp'],
```